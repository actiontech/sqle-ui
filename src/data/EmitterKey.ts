enum EmitterKey {
  Refresh_User_list = 'REFRESH_USER_LIST',
  Refresh_Role_list = 'REFRESH_ROLE_LIST',
  Refresh_Whitelist_List = 'REFRESH_WHITELIST_LIST',

  Refresh_Rule_Template_List = 'REFRESH_RULE_TEMPLATE_LIST',

  Reset_Create_Order_Form = 'RESET_CREATE_ORDER_FORM',

  Reset_Test_Data_Source_Connect = 'RESET_TEST_DATA_SOURCE_CONNECT',

  Reset_Workflow_Template_Form = 'RESET_WORKFLOW_TEMPLATE_FORM',

  Rest_Audit_Plan_Form = 'REST_AUDIT_PLAN_FORM',

  Refresh_Audit_Plan_Record = 'REFRESH_AUDIT_PLAN_RECORD',
}

export default EmitterKey;
