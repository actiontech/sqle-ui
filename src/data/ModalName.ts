export enum ModalName {
  // user manage
  Add_Role = 'ADD_ROLE',
  Update_Role = 'UPDATE_ROLE',

  Add_User = 'ADD_USER',
  Update_User = 'Update_User',
  Update_User_Password = 'Update_User_Password',

  // whitelist manage
  Add_Whitelist = 'ADD_WHITELIST',
  Update_Whitelist = 'UPDATE_WHITELIST',

  // order
  Modify_Sql = 'Modify_Sql',

  // account
  Modify_Password = 'Modify_Password',

  // rule template
  Clone_Rule_Template = 'CLONE_RULE_TEMPLATE',

  // header
  SHOW_VERSION = 'SHOW_VERSION',
}
