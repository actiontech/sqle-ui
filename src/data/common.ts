export enum ResponseCode {
  SUCCESS = 0,
}

export enum SystemRole {
  admin = 'admin',
}

export enum ModalSize {
  big = 1000,
  mid = 800,
}

export const RuleListDefaultTabKey = 'ALL';
export const instanceListDefaultKey = 'all';
export const ruleTemplateListDefaultKey = 'all';

export const ModalFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 7 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 13 },
  },
};

export const PageFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 8 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 12 },
    md: { span: 10 },
  },
};

export const PageBigFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 14 },
    // md: { span: 1 },
  },
};

export const FilterFormLayout = {
  labelCol: { span: 7 },
  wrapperCol: { span: 17 },
};

export const FilterFormColLayout = {
  xs: 24,
  sm: 12,
  xl: 8,
  xxl: 6,
};

export const filterFormButtonLayoutFactory = (
  smOffset = 0,
  xlOffset = 0,
  xxlOffset = 0
) => ({
  xs: 24,
  sm: {
    span: 12,
    offset: smOffset,
  },
  xl: {
    span: 8,
    offset: xlOffset,
  },
  xxl: {
    span: 6,
    offset: xxlOffset,
  },
});

export const FilterFormRowLayout = {
  gutter: 24,
};

export enum LoginTypeEnum{
  'ldap'='ldap',
  'sqle'='sqle',
}