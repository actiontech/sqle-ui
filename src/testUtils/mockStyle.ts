import * as useStyles from '../theme';

export const mockUseStyle = () => {
  const useStylesSpy = jest.spyOn(useStyles, 'default');
  useStylesSpy.mockImplementation(() => ({
    loginBg: 'loginBg-test',
    headerBg: 'headerBg-test',
    editor: 'editor-test',
  }));
  return useStylesSpy;
};
