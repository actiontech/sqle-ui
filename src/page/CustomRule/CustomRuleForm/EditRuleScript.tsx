import MonacoEditor, { MonacoEditorProps } from 'react-monaco-editor';
import { ComponentType, useEffect } from 'react';
import useChangeTheme from '../../../hooks/useChangeTheme';
import useMonacoEditor from '../../../hooks/useMonacoEditor';
import { EditRuleScriptFields, EditRuleScriptProps } from '.';
import { Button, Form } from 'antd';
import { useTranslation } from 'react-i18next';
import { PageFormLayout } from '../../../data/common';
import useStyles from '../../../theme';
import FooterButtonWrapper from '../../../components/FooterButtonWrapper';

const MonacoEditorFunComponent =
  MonacoEditor as ComponentType<MonacoEditorProps>;

const EditRuleScript: React.FC<EditRuleScriptProps> = (props) => {
  const { currentEditorTheme } = useChangeTheme();
  const { t } = useTranslation();
  const theme = useStyles();

  const { editorDidMount, registerRegexLanguage } = useMonacoEditor(
    props.form,
    {
      formName: 'script',
      placeholder: '',
    }
  );

  useEffect(() => {
    if (props.defaultData) {
      props.form.setFieldsValue({
        script: props.defaultData.rule_script,
      });
    }
  }, [props.defaultData, props.form]);

  return (
    <Form<EditRuleScriptFields> {...PageFormLayout} form={props.form}>
      <Form.Item
        name="script"
        label={t('customRule.editScriptForm.inputRuleScript')}
        wrapperCol={{
          ...PageFormLayout.wrapperCol,
          className: theme.editor,
        }}
        rules={[
          {
            required: true,
          },
        ]}
      >
        <MonacoEditorFunComponent
          theme={currentEditorTheme}
          width="100%"
          height="500"
          language="regexp"
          editorDidMount={(editor, monaco) => {
            editorDidMount(editor, monaco);
            registerRegexLanguage(editor, monaco);
          }}
          options={{
            automaticLayout: true,
          }}
        />
      </Form.Item>

      <FooterButtonWrapper insideProject={false}>
        <Button disabled={props.submitLoading} onClick={props.prevStep}>
          {t('common.prevStep')}
        </Button>
        <Button
          loading={props.submitLoading}
          onClick={props.submit}
          type="primary"
        >
          {t('common.submit')}
        </Button>
      </FooterButtonWrapper>
    </Form>
  );
};

export default EditRuleScript;
