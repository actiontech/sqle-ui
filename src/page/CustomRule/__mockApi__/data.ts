import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
