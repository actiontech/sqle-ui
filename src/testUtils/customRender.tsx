import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { Dictionary } from '../types/common.type';
import { storeFactory } from './mockRedux';
import lightTheme from '../theme/light';
import { ThemeProvider } from '@material-ui/styles';
import { mount, shallow } from 'enzyme';

type RenderParams = Parameters<typeof render>;
type MountParams = Parameters<typeof mount>;
type ShallowParams = Parameters<typeof shallow>;

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState),
    },
  } as any);
};

export const renderWithServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(<Router {...props}>{ui}</Router>, option);
};

export const renderWithMemoryRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(<MemoryRouter {...props}>{ui}</MemoryRouter>, option);
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(<ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>, option);
};

export const renderWithThemeAndRouter = (
  ...[ui, option]: [...RenderParams]
) => {
  return render(
    <MemoryRouter>
      <ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>
    </MemoryRouter>,
    option
  );
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme,
    },
  });
};

export const shallowWithTheme = (...[ui, option]: [...ShallowParams]) => {
  return shallow(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme,
    },
  });
};
