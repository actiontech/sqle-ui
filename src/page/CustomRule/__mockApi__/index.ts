import rule_template from '../../../api/rule_template';
import { resolveThreeSecond } from '../../../testUtils/mockRequest';
import { customRules } from './data';

export const mockGetCustomRules = () => {
  const spy = jest.spyOn(rule_template, 'getCustomRulesV1');
  spy.mockImplementation(() => resolveThreeSecond(customRules));
  return spy;
};

export const mockDeleteCustomRule = () => {
  const spy = jest.spyOn(rule_template, 'deleteCustomRuleV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};

export const mockCreateCustomRule = () => {
  const spy = jest.spyOn(rule_template, 'createCustomRuleV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};

export const mockUpdateCustomRule = () => {
  const spy = jest.spyOn(rule_template, 'updateCustomRuleV1');
  spy.mockImplementation(() => resolveThreeSecond({}));
  return spy;
};

export const mockGetCustomRule = () => {
  const spy = jest.spyOn(rule_template, 'getCustomRuleV1');
  spy.mockImplementation(() => resolveThreeSecond(customRules[0]));
  return spy;
};
