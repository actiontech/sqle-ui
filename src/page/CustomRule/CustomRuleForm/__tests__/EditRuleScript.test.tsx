/* eslint-disable no-console */
import { renderHook } from '@testing-library/react-hooks';
import { EditRuleScriptProps } from '..';
import { renderWithRedux } from '../../../../testUtils/customRender';
import { mockUseStyle } from '../../../../testUtils/mockStyle';
import EditRuleScript from '../EditRuleScript';
import { Form } from 'antd';
import { customRules } from '../../__mockApi__/data';
import { fireEvent, screen } from '@testing-library/react';

describe('test EditRuleScript', () => {
  const error = console.error;
  beforeAll(() => {
    console.error = jest.fn((message: any) => {
      if (
        message.includes(
          'A component is changing an uncontrolled input to be controlled'
        )
      ) {
        return;
      }
      error(message);
    });
  });

  afterAll(() => {
    console.error = error;
  });
  test('should match snapshot', () => {
    mockUseStyle();
    const { result } = renderHook(() => Form.useForm());
    const mockPreStep = jest.fn();
    const mockSubmit = jest.fn();
    const props: EditRuleScriptProps = {
      form: result.current[0],
      submitLoading: false,
      prevStep: mockPreStep,
      submit: mockSubmit,
      defaultData: customRules[0],
    };
    const { container } = renderWithRedux(
      <EditRuleScript {...props} />,
      undefined,
      {
        user: {
          theme: 'light',
        },
      }
    );

    expect(container).toMatchSnapshot();

    fireEvent.click(screen.getByText('common.prevStep'));
    expect(mockPreStep).toBeCalledTimes(1);

    fireEvent.click(screen.getByText('common.submit'));
    expect(mockPreStep).toBeCalledTimes(1);
  });
});
