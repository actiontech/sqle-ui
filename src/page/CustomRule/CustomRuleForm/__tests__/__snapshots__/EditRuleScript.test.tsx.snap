// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test EditRuleScript should match snapshot 1`] = `
<div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-form-item ant-form-item-has-success"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
        >
          <label
            class="ant-form-item-required"
            for="script"
            title="customRule.editScriptForm.inputRuleScript"
          >
            customRule.editScriptForm.inputRuleScript
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control editor-test ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                height="500"
                id="script"
                language="regexp"
                options="[object Object]"
                theme="vs"
                value="w+"
                width="100%"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <section
      class="footer-button-wrapper inside-the-project-footer-button-wrapper"
    >
      <div
        class="ant-card ant-card-bordered"
      >
        <div
          class="ant-card-body"
          style="padding-right: 24px; padding-top: 10px; padding-bottom: 10px;"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center full-width-element flex-end-horizontal"
          >
            <div
              class="ant-space-item"
              style="margin-right: 8px;"
            >
              <button
                class="ant-btn ant-btn-default"
                type="button"
              >
                <span>
                  common.prevStep
                </span>
              </button>
            </div>
            <div
              class="ant-space-item"
            >
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  common.submit
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </section>
  </form>
</div>
`;
